### PR TITLE
implement color functionality for material page

### PR DESCRIPTION
**Pull request by ivan-bonk**
color add, delete
disable color deleting, if color bound to material (show dialog window with material links)
add filter materials by colors

### CODE REVIEW

File: src/components/create-color/create-color.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      handleTabsChange(null, 0);
+    },
+    []
+  );
+
+  const tabPanels = languages.map((lang, index) => (
+    <TabPanel key={lang} value={tabsValue} index={index}>
+      <div className={styles.materialItemAdd}>
+        <TextField
+          data-cy={`${lang}Name`}
```
Comment (moran711) [inline src/components/create-color/create-color.js:101]: Please change naming to kebab case.

File: src/components/create-color/create-color.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+          error={touched[`${lang}Name`] && !!errors[`${lang}Name`]}
+          multiline
+          value={values[`${lang}Name`]}
+          onChange={handleChange}
+        />
+        {touched[`${lang}Name`] && errors[`${lang}Name`] && (
+          <div className={styles.inputError}>{errors[`${lang}Name`]}</div>
+        )}
+        <TextField
+          data-cy={`${lang}SimpleName`}
```
Comment (moran711) [inline src/components/create-color/create-color.js:115]: same

File: src/components/create-color/create-color.styles.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { makeStyles } from '@material-ui/core/styles';
```
Comment (moran711) [inline src/components/create-color/create-color.styles.js:1]: We have some of this styles in config folder.

File: src/redux/color/color.actions.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  SET_COLORS_ERROR,
+  ADD_COLOR_TO_STATE,
+  SHOW_COLOR_DIALOG_WINDOW,
+  SHOW_BOUND_MATERIALS_WINDOW,
+  SET_BOUND_MATERIALS
+} from './color.types';
+
+export const getColors = () => ({
+  type: GET_COLORS
+  // maybe make sense to add filters to backend
```
Comment (moran711) [inline src/redux/color/color.actions.js:19]: Please remove this commented line

File: src/components/colors-bar/colors-bar.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+const { SMALL_SIZE } = config.iconSizes;
+const { DEFAULT_CIRCLE, SMALL_CIRCLE } = config.colorCircleSizes;
+const { REMOVE_COLOR_MESSAGE } = config.messages;
+const { CREATE_COLOR_TITLE } = config.buttonTitles;
+const { createColorTitle, alreadyUse } = config.titles.colorTitles;
+const { name, mainLabel } = config.labels.color;
+
+function ColorsBar({ onColorChange, color }) {
+  const dispatch = useDispatch();
+  const { colors, boundMaterials, showBound, showCreateColor } = useSelector(
```
Comment (moran711) [inline src/components/colors-bar/colors-bar.js:32]: Please move this selector to separate file.

File: src/components/create-color/create-color.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+} = colorErrorMessages;
+const { CREATE_COLOR_TITLE } = buttonTitles;
+
+const { DEFAULT_CIRCLE } = config.colorCircleSizes;
+
+const CreateColor = () => {
+  const [colorPicker, setColorPicker] = useState(false);
+  const { createColor, tabsValue, handleTabsChange } = useColorHandlers();
+
+  const { loading } = useSelector(({ Color }) => ({
```
Comment (moran711) [inline src/components/create-color/create-color.js:31]: Please move this selector to separate file.

File: src/components/material-form/material-form.js [hunk 7]
```diff
@@ -47,20 +29,11 @@ function MaterialForm({ material, id }) {
   const styles = useStyles();
   const dispatch = useDispatch();
 
-  const { loading, colors } = useSelector(({ Material }) => ({
-    loading: Material.materialLoading,
-    colors: Material.colors
+  const { loading } = useSelector(({ Material }) => ({
```
Comment (moran711) [inline src/components/material-form/material-form.js:32]: Please move this selector to separate file.

File: src/pages/material/material-page/material-page.js [hunk 8]
```diff
@@ -33,29 +42,38 @@ const MaterialPage = () => {
   const { openSuccessSnackbar } = useSuccessSnackbar();
   const {
     list,
+    filter,
     loading,
     pagesCount,
     currentPage,
     materialsPerPage
   } = useSelector(({ Material }) => ({
```
Comment (moran711) [inline src/pages/material/material-page/material-page.js:50]: Please move this selector to separate file.

File: src/pages/material/material-page/material-page.js [hunk 9]
```diff
@@ -33,29 +42,38 @@ const MaterialPage = () => {
...
     list: Material.list,
+    filter: Material.filter,
     loading: Material.materialLoading,
     pagesCount: Material.pagination.pagesCount,
     currentPage: Material.pagination.currentPage,
     materialsPerPage: Material.pagination.materialsPerPage
   }));
 
+  const colors = useSelector(({ Color }) => Color.list);
```
Comment (moran711) [inline src/pages/material/material-page/material-page.js:59]: Why do you need second selector?

**Review by kapoliub — APPROVED** [review]

**Review by Iryna-Bzdyr — APPROVED** [review]

**Review by moran711 — APPROVED** [review]

**Pull request merged**